wherehows-web/app/routes/search.ts

```diff
@@ -24,10 +24,10 @@ export default class SearchRoute extends Route.extend(AuthenticatedRouteMixin) {
    * Add spinner when model is loading
    */
   @action
-  loading(transition: any) {
+  loading(transition: import('ember').Ember.Transition): void {
     let controller = this.controllerFor('search');
     controller.set('searchLoading', true);
-    transition.promise.finally(function() {
+    transition.promise!.finally(function() {
```

Comment:
What is this?

Reply:
[Definite Assignment Assertion](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-2-7.html) The type def for Transition.promise as [per](https://github.com/tildeio/router.js/blob/e853a63d2efd7337c49e177baab244bd0604b5c8/lib/router/transition.ts#L43) , (although it's incorrect and augmented [here](https://github.com/linkedin/WhereHows/pull/1394/files#diff-dda9c050f67113ed663ec24ed01597d0R43)) is optional. Since you aren't checking to see if it's there I'm guessing you expect it to be there for this use case. Otherwise you can insert a safety check before